src/datamigration/azext_datamigration/manual/custom.py

```diff
@@ -0,0 +1,245 @@
+        logFilePath = os.path.join(defaultOutputFolder, "Logs")
+        print(f"Event and Error Logs Folder Path: {logFilePath}")
+
+    except Exception as e:
+        raise e
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Assessment helper function to test whether the given cofig_file_path is valid and has valid action specified.
+# -----------------------------------------------------------------------------------------------------------------
+def test_config_file_path(path):
+
+    if not os.path.exists(path):
+        raise CLIError(f'Invalid config file path: {path}. Please provide a valid config file path.')
+
+    # JSON file
+    with open(path, "r", encoding=None) as f:
+        configJson = json.loads(f.read())
+
+    if not ((configJson['action'] == "Assess") or (configJson['action'] == "assess")):
+        raise CLIError("The desired action in config file was invalid. Please use \"Assess\" for action property in config file")
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Assessment helper function to test whether the given cofig_file_path is valid and has valid action specified.
+# -----------------------------------------------------------------------------------------------------------------
+def get_default_output_folder():
+
+    osPlatform = platform.system()
+
+    if osPlatform.__contains__('Linux'):
+        defaultOutputPath = os.path.join(os.getenv('USERPROFILE'), ".config", "Microsoft", "SqlAssessment")
+    elif osPlatform.__contains__('Darwin'):
+        defaultOutputPath = os.path.join(os.getenv('USERPROFILE'), "Library", "Application Support", "Microsoft", "SqlAssessment")
+    else:
+        defaultOutputPath = os.path.join(os.getenv('LOCALAPPDATA'), "Microsoft", "SqlAssessment")
+
+    return defaultOutputPath
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Register Sql Migration Service on IR command Implementation.
+# -----------------------------------------------------------------------------------------------------------------
+def datamigration_register_ir(auth_key,
+                              ir_path=None):
+
+    osPlatform = platform.system()
+
+    if not osPlatform.__contains__('Windows'):
+        raise CLIError("This command cannot be run in non-windows environment. Please run this command in Windows environement")
+
+    if not is_user_admin():
+        raise CLIError("Failed: You do not have Administrator rights to run this command. Please re-run this command as an Administrator!")
+    validate_input(auth_key)
+    if ir_path is not None:
+        if not os.path.exists(ir_path):
+            raise CLIError(f"Invalid gateway path : {ir_path}. Please provide a valid Integration Runtime MSI path")
+        install_gateway(ir_path)
+
+    register_ir(auth_key)
```

Comment:
IR registration can be done only if we have the right IR path, right?

Reply:
Yes, ir_path is for installation of IR. ir_path here is the installer path. Once the installation is done register of auth key is done. If no IR is found, an CLIError is raised.